src/integration/integration_api_broker_graceful_shutdown_test.go

```diff
@@ -52,17 +62,24 @@ var _ = Describe("Integration_Api_Broker_Graceful_Shutdown", func() {
 				newConn.Close()
 
 				sendSigusr2Signal(APIServer)
+				Eventually(buffer, 5*time.Second).Should(gbytes.Say(`Received SIGUSR2 signal`))
+
+				Eventually(processMap[APIServer].Wait()).Should(Receive())
+				Expect(runner.ExitCode()).Should(Equal(0))
 
 				_, err = net.Dial("tcp", fmt.Sprintf("127.0.0.1:%d", components.Ports[APIServer]))
 				Expect(err).To(HaveOccurred())
 
 			})
 
-			It("waits for in-flight request to finish before shutting down", func() {
+			It("waits for in-flight request to finish before shutting down server on receiving SIGUSR2 signal", func() {
 				conn, err := net.Dial("tcp", fmt.Sprintf("127.0.0.1:%d", components.Ports[APIServer]))
 				Expect(err).NotTo(HaveOccurred())
 
 				sendSigusr2Signal(APIServer)
+				Eventually(buffer, 5*time.Second).Should(gbytes.Say(`Received SIGUSR2 signal`))
+
+				Expect(runner.ExitCode()).Should(Equal(-1))
```

Comment:
Something isn't right here.
I added the following 
				Eventually(runner.ExitCode).Should(BeZero())
and it did exit which is not expected.